crates/lox-io/src/ndm/opm.rs

```diff
@@ -0,0 +1,968 @@
+//! #         <MAN_DV_2 units="km/s">0.270000</MAN_DV_2>
+//! #         <MAN_DV_3 units="km/s">0.000000</MAN_DV_3>
+//! #     </maneuverParameters>
+//! # </data>
+//! # </segment>
+//! # </body>
+//! # </opm>"#;
+//! #
+//! # use lox_io::ndm::opm::OpmType;
+//! #
+//! let message: OpmType = quick_xml::de::from_str(xml).unwrap();
+//! ```
+//!
+//! To deserialize a KVN message:
+//! ```
+//! # let kvn = r#"CCSDS_OPM_VERS = 3.0
+//! # COMMENT Generated by GSOC, R. Kiehling
+//! # COMMENT Current intermediate orbit IO2 and maneuver planning data
+//! # CREATION_DATE = 2021-06-03T05:33:00.123
+//! # ORIGINATOR = GSOC
+//! # OBJECT_NAME = EUTELSAT W4
+//! # OBJECT_ID = 2021-028A
+//! # CENTER_NAME = EARTH
+//! # REF_FRAME = TOD
+//! # TIME_SYSTEM = UTC
+//! # COMMENT State Vector
+//! # EPOCH = 2021-06-03T00:00:00.000
+//! # X = 6655.9942 [km]
+//! # Y = -40218.5751 [km]
+//! # Z = -82.9177 [km]
+//! # X_DOT = 3.11548208 [km/s]
+//! # Y_DOT = 0.47042605 [km/s]
+//! # Z_DOT = -0.00101495 [km/s]
+//! # COMMENT Keplerian elements
+//! # SEMI_MAJOR_AXIS = 41399.5123 [km]
+//! # ECCENTRICITY = 0.020842611
+//! # INCLINATION = 0.117746 [deg]
+//! # RA_OF_ASC_NODE = 17.604721 [deg]
+//! # ARG_OF_PERICENTER = 218.242943 [deg]
+//! # TRUE_ANOMALY = 41.922339 [deg]
+//! # GM = 398600.4415 [km**3/s**2]
+//! # COMMENT Spacecraft parameters
+//! # MASS = 1913.000 [kg]
+//! # SOLAR_RAD_AREA = 10.000 [m**2]
+//! # SOLAR_RAD_COEFF = 1.300
+//! # DRAG_AREA = 10.000 [m**2]
+//! # DRAG_COEFF = 2.300
+//! # COMMENT 2 planned maneuvers
+//! # COMMENT First maneuver: AMF-3
+//! # COMMENT Non-impulsive, thrust direction fixed in inertial frame
+//! # MAN_EPOCH_IGNITION = 2021-06-03T09:00:34.1
+//! # MAN_DURATION = 132.60 [s]
+//! # MAN_DELTA_MASS = -18.418 [kg]
+//! # MAN_REF_FRAME = EME2000
+//! # MAN_DV_1 = -0.02325700 [km/s]
+//! # MAN_DV_2 = 0.01683160 [km/s]
+//! # MAN_DV_3 = -0.00893444 [km/s]
+//! # COMMENT Second maneuver: first station acquisition maneuver
+//! # COMMENT impulsive, thrust direction fixed in RTN frame
+//! # MAN_EPOCH_IGNITION = 2021-06-05T18:59:21.0
+//! # MAN_DURATION = 0.00 [s]
+//! # MAN_DELTA_MASS = -1.469 [kg]
+//! # MAN_REF_FRAME = RTN
+//! # MAN_DV_1 = 0.00101500 [km/s]
+//! # MAN_DV_2 = -0.00187300 [km/s]
+//! # MAN_DV_3 = 0.00000000 [km/s]"#;
+//! #
+//! # use lox_io::ndm::kvn::KvnDeserializer;
+//! # use lox_io::ndm::opm::OpmType;
+//! #
+//! let message: OpmType = KvnDeserializer::deserialize(&mut kvn.lines().peekable()).unwrap();
+//! ```
+
+use serde;
+
+use super::common;
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct OpmType {
+    #[serde(rename = "@id")]
+    // Marked as option for the KVN deserializer
+    pub id: Option<String>,
+    #[serde(rename = "@version")]
+    pub version: String,
+    #[serde(rename = "header")]
+    pub header: common::OdmHeader,
```

Comment:
It's just automatically generated by `xml-schema-derive`. I'd keep it like this just for uniformity, if that's okay with you.

Reply:
Ah! I foolishly thought this was all manual! Yes, no need to be fiddling with generated code.

Could you please add header comments to all the generated files explaining what's generated, how it's generated, and how it can be re-generated if required?

Reply:
It's not all generated. I have done some adjustments to it, especially to get it to work with KVN. I will add a comment for this.